credentials/credentials.go

```diff
@@ -84,9 +92,38 @@ func (c *Credentials) Decrypt(input io.Reader) (io.Reader, error) {
 	return out, nil
 }
 
+func (c *Credentials) EncryptV2(data []byte) ([]byte, error) {
+	nonce := make([]byte, c.aead.NonceSize(), c.aead.NonceSize()+len(data)+c.aead.Overhead())
+	if _, err := rand.Read(nonce); err != nil {
+		return nil, fmt.Errorf("failed to generate random nonce: %w", err)
+	}
+
+	encrypted := c.aead.Seal(nonce, nonce, data, nil)
+	return encrypted, nil
+}
+
+func (c *Credentials) DecryptV2(ciphertext []byte) ([]byte, error) {
+	nonce, message := ciphertext[:c.aead.NonceSize()], ciphertext[c.aead.NonceSize():]
+	decrypted, err := c.aead.Open(nil, nonce, message, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decrypt: %w", err)
+	}
+
+	return decrypted, nil
+}
+
 func (c *Credentials) HashString(value string) string {
 	h := sha256.New()
 	_, _ = fmt.Fprintf(h, "%s:%s", c.recipient, value)
 
 	return hex.EncodeToString(h.Sum(nil))
 }
+
+func (c *Credentials) HashStrings(data ...string) []byte {
```

Comment:
This doesn't reference `c` at all so it doesn't really make sense for it to be a method on `Credentials`, it could just be a utility function.